service/src/db/dbOperations.ts

```diff
@@ -174,10 +182,22 @@ export async function batchUpdate(artifacts: FhirArtifact[], action: string) {
     console.log(`Batch ${action} transaction committed.`);
   } catch (err) {
     console.log(`Batch ${action} transaction failed: ` + err);
-    error = err;
+    error = handlePossibleDuplicateKeyError(err);
   } finally {
     await updateSession?.endSession();
   }
   if (error) throw error;
   return results;
 }
+
+function handlePossibleDuplicateKeyError(e: any) {
+  if (e instanceof MongoServerError && e.code === 11000) {
+    let errorString = 'Resource with primary identifiers already in repository.';
+    if (e.keyPattern) {
+      errorString =
+        'Resource with identifiers (' + Object.keys(e.keyPattern).join(',') + ') already exists in the repository.';
```

Comment:
Should the error message identify what url/version are duplicating? This might help especially when multiple things are being inserted/updated in succession.
For instance, the dup key error looks something like this: 
"Unexpected: E11000 duplicate key error collection: measure-repository.Measure index: url_1_version_1 dup key: { url: \"SevereObstetricComplicationsFHIR-test\", version: \"SevereObstetricComplicationsFHIR\" }"
We could even stick the mongo error message in as part of our new error string.

Reply:
Added the key info and the resource type to the error message.